osu.Game/Screens/Play/HUD/ArgonSongProgressBar.cs

```diff
@@ -33,6 +35,12 @@ public partial class ArgonSongProgressBar : SongProgressBar
 
         private double trackTime => (EndTime - StartTime) * Progress;
 
+        private float relativePositionX;
+
+        public LocalisableString TooltipText => $"{formatTime(TimeSpan.FromSeconds(relativePositionX > 0 ? Math.Round((EndTime - StartTime) * relativePositionX / DrawWidth / 1000)
+            : relativePositionX > DrawWidth ? Math.Round(EndTime / 1000) : 0))}"
+                                                + $" - {(relativePositionX > 0 ? Math.Round(relativePositionX / DrawWidth * 100, 1) : relativePositionX > DrawWidth ? 100 : 0)}%";
```

Comment:
Holy heck this is unreadable. Nested ternaries? Everything inlined? Why though?

Please split this to an actual method with actual variables thanks.

Also for percentage formatting you should be using the [`P` standard format spec](https://learn.microsoft.com/en-us/dotnet/standard/base-types/standard-numeric-format-strings) rather than rolling your own.

Reply:
Just fixed this in the latest commit.
Even I couldn't understand my code written few days ago 🤣